conformance/lang/expressions/literals/int-literal.balt

```diff
@@ -0,0 +1,470 @@
+    io:println(y is decimal); // @output false
+
+    any z = 0x12aB;
+    io:println(z is int); // @output true
+    io:println(z is float); // @output false
+    io:println(z is decimal); // @output false
+}
+
+Test-Case: error
+Description: Test the basic type when the applicable contextually expected type is not a subtype of either decimal or
+             float and the numeric literal is an int-literal.
+Fail-Issue: ballerina-lang #32698
+Labels: decimal, float, int
+
+public function main() {
+    float|decimal a = 1; // @error expected a '(float|decimal)' but found a 'int'
+}
+
+Test-Case: output
+Description: Test the basic type when the applicable contextually expected type is not a subtype of either decimal or
+             float and the numeric literal is not an int-literal.
+Labels: any, decimal, float, int, is-expr
+
+public function main() {
+    int|float|decimal x = 4311.1;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    float|decimal y = 932.32e11;
+    io:println(y is float); // @output true
+    io:println(y is decimal); // @output false
+
+    any z = 123.654;
+    io:println(z is int); // @output false
+    io:println(z is float); // @output true
+    io:println(z is decimal); // @output false
+}
+
+Test-Case: output
+Description: Test int literals with valid type casts.
+Labels: decimal, float, int, type-cast-expr
+
+public function main() {
+    io:println(<int> 13245); // @output 13245
+
+    io:println(<int> 0xabde12a); // @output 180216106
+
+    io:println(<decimal> 13245); // @output 13245
+
+    io:println(<decimal> 0xabde12a); // @output 180216106.0
+
+    io:println(<float> 13245); // @output 13245.0
+
+    io:println(<float> 0xabde12a); // @output 1.80216106E8
+}
+
+Test-Case: error
+Description: Test nil literal '()' with invalid type casts.
+Fail-Issue: ballerina-lang #32716
+Labels: any, boolean, int, nil-literal, string, type-cast-expr
+
+public function main() {
+    any x = <boolean> 0; // @error 'int' cannot be cast to 'boolean'
+
+    x = <boolean> 0x0; // @error 'int' cannot be cast to 'boolean'
+
+    x = <()> 123; // @error 'int' cannot be cast to '()'
+
+    x = <()> 0X42af; // @error 'int' cannot be cast to '()'
+
+    x = <string> 456; // @error 'int' cannot be cast to 'string'
+
+    x = <string> 0xa21bd; // @error 'int' cannot be cast to 'string'
+}
+
+Test-Case: output
+Description: Test int literals with interpolation in template expressions.
+Labels: int, raw-template-expr
+
+public function main() {
+    io:println(`${124}`); // @output 124
+
+    io:println(`${0x7ab1}`); // @output 31409
+}
+
+Test-Case: output
+Description: Test int literals as member of structural constructors.
+Labels: array-type, int, map-type
+
+public function main() {
+    io:println([123, 0x12a3]); // @output [123,4771]
+
+    io:println({"c" : 987, "b" : 0X92c1}); // @output {"c":987,"b":37569}
+}
```

Comment:
There is no type involved here. Let's add labels `mapping-constructor-expr`, `list-constructor-expr` and remove type labels